app/models/user.rb

```diff
@@ -25,80 +29,165 @@ def to_s
+    @groups ||= begin
+      perm_keys = ["sunetid:#{login}"]
+      return perm_keys unless webauth && webauth.privgroup.present?
+      perm_keys + webauth.privgroup.split(/\|/).map {|g| "workgroup:#{g}"}
+    end
   end
 
   # @return [Boolean] is the user a repository wide administrator
   def is_admin
-    !(groups & ADMIN_GROUPS).empty?
-  end
-
-  # @return [Boolean] is the user a repository wide viewer
-  def is_viewer
-    !(groups & VIEWER_GROUPS).empty?
+    !(groups & ADMIN_GROUPS).blank?
   end
 
   # @return [Boolean] is the user a repo wide manager
   def is_manager
-    !(groups & MANAGER_GROUPS).empty?
+    !(groups & MANAGER_GROUPS).blank?
+  end
+
+  # @return [Boolean] is the user a repository wide viewer
+  def is_viewer
+    !(groups & VIEWER_GROUPS).blank?
   end
 
   # The convention is for boolean is_XYZ methods to be interrogative (end in "?")
   alias_method :is_admin?,   :is_admin
-  alias_method :is_viewer?,  :is_viewer
   alias_method :is_manager?, :is_manager
+  alias_method :is_viewer?,  :is_viewer
 
   def login
     webauth ? webauth.login : sunetid
   end
 
-  ##
+  # Validate access permissions using Argo user roles. If that fails, check the
+  # APO role permissions.  If the object is an APO, first check permissions on
+  # the governing APO of this APO object. As a last resort, check the object
+  # roles when this object is an APO.
+  # @param dor_object [String|Dor::Base] Accepts a DRUID string or
+  #                                      any subclass of Dor::Base
+  # @return [Boolean]
+  def can_manage_object?(dor_object)
+    dor_object = parse_object(dor_object)
+    pid = dor_object.pid
+    apo_pid = dor_object.admin_policy_object.pid
+    manager_access = is_admin || is_manager
+    manager_access ||= dor_object.can_manage_item?(roles(apo_pid))
+    manager_access ||= dor_object.can_manage_content?(roles(apo_pid))
+    manager_access ||= dor_object.is_a?(Dor::AdminPolicyObject) && dor_object.can_manage_item?(roles(pid))
+    manager_access ||= dor_object.is_a?(Dor::AdminPolicyObject) && dor_object.can_manage_content?(roles(pid))
+    manager_access
+  end
+
+  # Validate access permissions using Argo user roles. If that fails, check the
+  # APO role permissions.  If the object is an APO, first check permissions on
+  # the governing APO of this APO object. As a last resort, check the object
+  # roles when this object is an APO.
+  # @param dor_object [String|Dor::Base] Accepts a DRUID string or
+  #                                      any subclass of Dor::Base
+  # @return [Boolean]
+  def can_view_object?(dor_object)
+    dor_object = parse_object(dor_object)
+    return true if can_manage_object?(dor_object)
+    pid = dor_object.pid
+    apo_pid = dor_object.admin_policy_object.pid
+    view_access = is_viewer
+    view_access ||= dor_object.can_view_content?(roles(apo_pid))
+    view_access ||= dor_object.is_a?(Dor::AdminPolicyObject) && dor_object.can_view_content?(roles(pid))
```

Comment:
it appears that this line implements the behavior of allowing a user to view an APO which grants them access to view other objects, even if they wouldn't otherwise have the right to view that governing APO.  however, this logic also appears to be implemented in `allow_view` in items_controller.rb.  why not just call `can_view_object` over there?  this doesn't actually appear to be called at all.